Tests/Tests/APIClientSpec.swift

```diff
@@ -191,6 +191,50 @@ class APIClientSpec: QuickSpec {
           expect(error).toEventually(equal(SwishError.ServerError(code: expectedCode, json: expectedJSON)))
         }
       }
+
+      describe("dispatch queue configuration") {
+        it("dispatches onto the main queue by default") {
+          let performer = FakeRequestPerformer(responseData: .JSON([:]))
+          let client = APIClient(requestPerformer: performer)
+
+          var isOnMain: Bool?
+
+          client.performRequest(FakeRequest()) { _ in
+            isOnMain = NSThread.isMainThread()
+          }
+
+          expect(isOnMain).toEventually(beTrue())
+        }
+
+        it("doesn't dispatch onto main queue if queue is set to nil") {
+          let performer = FakeRequestPerformer(responseData: .JSON([:]), background: true)
+          var client = APIClient(requestPerformer: performer)
+          client.queue = nil
+
+          var isOnMain: Bool?
+
+          client.performRequest(FakeRequest()) { _ in
+            isOnMain = NSThread.isMainThread()
+          }
+
+          expect(isOnMain).toEventually(beFalse())
```

Comment:
I don't think this actually tests anything. `isOnMain` is `false` by default, so this can't fail. Would probably be better to default it to `true` and remove the optionality.